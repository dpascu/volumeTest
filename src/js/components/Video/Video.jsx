import React from 'react';
import PropTypes from 'prop-types';

/**
 * Video component
 * @param {Object} props
 */
const Video = ({ src }) => (
  <video autoPlay className="video-full">
    <track kind="captions" />
    <source src={src} type="video/mp4" />
    Your browser does not support the video tag.
  </video>
);

Video.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Video;
