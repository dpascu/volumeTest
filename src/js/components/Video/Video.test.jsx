/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';

import Video from './Video';

const SRC_VIDEO = '../../../media/video.mp4';

describe('Component Video', () => {
  it('should render component', () => {
    const component = renderer.create(
      <Video src={SRC_VIDEO} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
