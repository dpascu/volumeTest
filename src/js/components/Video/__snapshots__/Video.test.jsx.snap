// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Video should render component 1`] = `
<video
  autoPlay={true}
  className="video-full"
>
  <track
    kind="captions"
  />
  <source
    src="../../../media/video.mp4"
    type="video/mp4"
  />
  Your browser does not support the video tag.
</video>
`;
